### DISCUSSION TITLE
ArgumentOutOfRangeException on analyzer

### DISCUSSION THREAD

**Discussion by mosphat** [xunit/xunit#2823] [Question] [answered]
Hello,

On one of my projects the Xunit analyzer throws this error: Analyzer 'Xunit.Analyzers.InlineDataMustMatchTheoryParameters' threw an exception of type 'System.ArgumentOutOfRangeException' with message 'Index was out of range. Must be non-negative and less than the size of the collection.

My guess is there's an invalid test somewhere, but given the sheer number of tests in that project it's a rather daunting task to comment them one by one until the error goes away. Is it possible to run the analyzer in debug mode somehow so I can see on what test it fails? Or perhaps more useful: is it possible to include the test name in the error message?

**Comment by bartelink** [comment]
If you have any more context such as xunit (and analyzer) versions, and ideally a full stack trace, it will help people to provide you a more relevant answer and/or reproduce the problem.

**Reply by bradwilson** [reply]
Yes, we will need a repro of _your code_ that's causing the error so that we can track it down. Thanks!

**Comment by mosphat** [comment]
I'm using xunit 2.6.2, xunit.analyzers 1.6.0 and xunit.runner.visualstudio 2.5.4, all latest stables version atm. The problem is the error shows up as a warning only sometimes. Currently it doesn't. No idea why.

Given its unpredictable nature and the large project, it's extremely time consuming to start commenting out tests in order to find the culprit. Hence my question on how to run things in debug mode so I can find the test that's causing it (and also get a stacktrace).

Posting a repro of my code is not an option. It's proprietary and even then far too large to be meaningful. Note that the error mentioned in the opening post is the only information I have. It doesn't mention a specific test or file.

**Reply by bradwilson** [reply]
There isn't really a good strategy for debugging analyzers. The system catches exceptions (and disables the analyzer), so there isn't a good chance for attaching a debugger.

Looking at the source, I only see one set of collection indexer usage and it should be well protected by the `for` statement: https://github.com/xunit/xunit.analyzers/blob/5c1edb6cbb2f987a7160641d9fabf7b1d9411f7c/src/xunit.analyzers/X1000/InlineDataMustMatchTheoryParameters.cs#L91-L94

**Reply by bradwilson** [reply]
Now I see three others (all the same kind of reference) which is a candidate:

https://github.com/xunit/xunit.analyzers/blob/5c1edb6cbb2f987a7160641d9fabf7b1d9411f7c/src/xunit.analyzers/X1000/InlineDataMustMatchTheoryParameters.cs#L141

https://github.com/xunit/xunit.analyzers/blob/5c1edb6cbb2f987a7160641d9fabf7b1d9411f7c/src/xunit.analyzers/X1000/InlineDataMustMatchTheoryParameters.cs#L167

https://github.com/xunit/xunit.analyzers/blob/5c1edb6cbb2f987a7160641d9fabf7b1d9411f7c/src/xunit.analyzers/X1000/InlineDataMustMatchTheoryParameters.cs#L192

**Reply by bradwilson** [reply, marked as answer]
Commit to fix it, if this is where it is: https://github.com/xunit/xunit.analyzers/commit/08daceb06a277a4879d5cb40468ddb04c1969039

Available in `1.7.0-pre.15`

**Reply by mosphat** [reply]
Thanks! Let's assume that indeed took care of it. If not, I'll post back here.
